### ISSUE TITLE
Conflict with "cordova-plugin-advanced-http" plugin 

### ISSUE THREAD

**Issue by microsmsm** [dpa99c/cordova-plugin-firebasex#601]
# Bug report

**CHECKLIST**
- [x] I have read the [issue reporting guidelines](https://github.com/dpa99c/cordova-plugin-firebasex#reporting-issues)

- [x] I confirm this is a suspected bug or issue that will affect other users

- [ ] I have reproduced the issue using the [example project](https://github.com/dpa99c/cordova-plugin-firebasex-test) or provided the necessary information to reproduce the issue.

- [x] I have read [the documentation](https://github.com/dpa99c/cordova-plugin-firebasex) thoroughly and it does not help solve my issue.

- [x] I have checked that no similar issues (open or closed) already exist.

**Current behavior:**
no made http requests after the very first time
logcat:
```
03-19 15:10:36.932 10921 11216 E AndroidRuntime: Process: com.microsmsm.microsmsm3, PID: 10921
03-19 15:10:36.932 10921 11216 E AndroidRuntime: java.lang.NoSuchMethodError: No virtual method getHttpUrlChecked(Ljava/lang/String;)Lokhttp3/HttpUrl; in class Lokhttp3/internal/Internal; or its super classes (declaration of 'okhttp3.internal.Internal' appears in /data/app/~~cPpaVobTEjG8ylSDBAnluQ==/com.microsmsm.microsmsm3-q_1Zq7ZF8oXMItemI_K5Hg==/base.apk!classes3.dex)
03-19 15:10:36.932 10921 11216 E AndroidRuntime: 	at okhttp3.internal.huc.OkHttpURLConnection.buildCall(OkHttpURLConnection.java:373)
03-19 15:10:36.932 10921 11216 E AndroidRuntime: 	at okhttp3.internal.huc.OkHttpURLConnection.getResponse(OkHttpURLConnection.java:435)
03-19 15:10:36.932 10921 11216 E AndroidRuntime: 	at okhttp3.internal.huc.OkHttpURLConnection.getResponseCode(OkHttpURLConnection.java:491)
03-19 15:10:36.932 10921 11216 E AndroidRuntime: 	at okhttp3.internal.huc.DelegatingHttpsURLConnection.getResponseCode(DelegatingHttpsURLConnection.java:106)
03-19 15:10:36.932 10921 11216 E AndroidRuntime: 	at okhttp3.internal.huc.OkHttpsURLConnection.getResponseCode(OkHttpsURLConnection.java:26)
03-19 15:10:36.932 10921 11216 E AndroidRuntime: 	at com.silkimen.http.HttpRequest.code(HttpRequest.java:1447)
03-19 15:10:36.932 10921 11216 E AndroidRuntime: 	at com.silkimen.http.HttpRequest.stream(HttpRequest.java:1740)
03-19 15:10:36.932 10921 11216 E AndroidRuntime: 	at com.silkimen.http.HttpRequest.buffer(HttpRequest.java:1729)
03-19 15:10:36.932 10921 11216 E AndroidRuntime: 	at com.silkimen.http.HttpRequest.receive(HttpRequest.java:1856)
03-19 15:10:36.932 10921 11216 E AndroidRuntime: 	at com.silkimen.cordovahttp.CordovaHttpBase.processResponse(CordovaHttpBase.java:189)
03-19 15:10:36.932 10921 11216 E AndroidRuntime: 	at com.silkimen.cordovahttp.CordovaHttpBase.run(CordovaHttpBase.java:82)
03-19 15:10:36.932 10921 11216 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
03-19 15:10:36.932 10921 11216 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
```

I think it's OKHTTP conflict

**Expected behavior:**

App working without problems making http requests 
**Steps to reproduce:**

install both plugins and make http reuqest
**Screenshots**

**Environment information**

- Cordova CLI version 
	- `10.0.0`
- Cordova platform version
	- `Installed platforms:
  android 9.0.0
Available platforms: 
  browser ^6.0.0
  electron ^1.0.0
  ios ^6.1.0
  osx ^6.0.0`
- Plugins & versions installed in project (including this plugin)
    - `cordova-android-play-services-gradle-release 4.0.0 "cordova-android-play-services-gradle-release"
cordova-plugin-advanced-http 2.5.1 "Advanced HTTP plugin"
cordova-plugin-androidx-adapter 1.1.3 "cordova-plugin-androidx-adapter"
cordova-plugin-camera 4.1.0 "Camera"
cordova-plugin-device 2.0.3 "Device"
cordova-plugin-file 6.0.2 "File"
cordova-plugin-firebasex 11.0.3-cli "Google Firebase Plugin"
cordova-plugin-google-analytics 1.9.0 "Google Universal Analytics Plugin"
cordova-plugin-googlemaps 2.7.1 "cordova-plugin-googlemaps"
cordova-plugin-ionic-keyboard 2.2.0 "cordova-plugin-ionic-keyboard"
cordova-plugin-ionic-webview 5.0.0 "cordova-plugin-ionic-webview"
cordova-plugin-okhttp 2.0.0-dev "OkHttp"
cordova-plugin-screen-orientation 3.0.2 "Screen Orientation"
cordova-plugin-splashscreen 5.0.4 "Splashscreen"
cordova-plugin-statusbar 2.4.3 "StatusBar"
cordova-plugin-whitelist 1.3.4 "Whitelist"
cordova-sqlite-storage 4.0.0 "Cordova sqlite storage plugin - cordova-sqlite-storage plugin version"
es6-promise-plugin 4.2.2 "Promise"`
- Dev machine OS and version, e.g.
    - OSX
        
_Runtime issue_
- Android 

**Related code:**
```
insert any relevant code here such as plugin API calls / input parameters
```

**Console output**
<details>
<summary>console output</summary>

```

// Paste any relevant JS/native console output here

```

</details><br/><br/>

**Other information:**

**Comment by dpa99c** [comment]
The OKHTTP library is a downstream dependency of the Firebase SDK components on Android.
i.e. this plugin does not directly depend on it or set a version dependency.
Therefore the conflict is between the Firebase SDK and `cordova-plugin-advanced-http` - i.e. if you removed this plugin and integrated the Firebase SDK directly into your Android Studio project, you would encounter the same problem.
Therefore there is nothing that can be done directly from within the scope of this plugin to resolve your conflict - you'll need to figure out what version of OKHTTP is being used by Firebase and how that conflicts with the other plugin.
So closing this as this plugin is not the cause nor can it resolve the issue.

**Comment by microsmsm** [comment]
@dpa99c can you please give me a hint how to figure out what version of OKHTTP is being used by Firebase?

**Comment by adamk22** [comment]
I ran into this same issue when I updated this plugin to the latest version. In my case it was also the cordova-plugin-advanced-http plugin that conflicted, specifically the okhttp3 library. 

In my case I was able to fix it by locking the version by adding the following to the config.xml:

```
  <preference name="AndroidXEnabled" value="true" />
  <preference name="OKHTTP_VERSION" value="3.12.1" />
```

I also added the Android X preference considering I didn't have that preference in my config as well. I'm sure it will also work without it.

**Comment by microsmsm** [comment]
I solved it too by adding the version 3.12.1 as variable while installing the plugin 
`ionic cordova plugin add cordova-plugin-firebasex --variable OKHTTP_VERSION=3.12.1`
![Screenshot_20210401-004621_Chrome](https://user-images.githubusercontent.com/7149686/113220716-f85abf80-9283-11eb-96a6-c88c1fefe7e6.jpg)
